### ISSUE TITLE
You have triggered an abuse detection mechanism

### ISSUE THREAD

**Issue by SpainTrain** [Jimdo/github-sync-labels-milestones#14]
We are consistently seeing this every time we run GSLM:

```
{
  "documentation_url": "https://developer.github.com/v3/#abuse-rate-limits",
  "message": "You have triggered an abuse detection mechanism. Please wait a few minutes before you try again."
}
```

My guess is that the cross product of repos to labels/milestones has grow large enough that it always triggers the quota.  Retrying the entire run doesn't work of course, because it retries the entire sync.

Would you be open to adding retries with exponential back-off to the requests, using something like https://www.npmjs.com/package/promise-retry or https://www.npmjs.com/package/retry-as-promised ?

**Comment by SpainTrain** [comment]
related: https://github.com/mikedeboer/node-github/issues/526

**Comment by Xiphe** [comment]
Hey @SpainTrain 

thanks for the feedback. That's what I call a success disaster ;)

I don't know how big your configuration is, but I agree that most likely you're hitting some kind of API request quota (I think most of GitHubs API have s.th. like 5k requests per hour).

I'd be open for adding retries. Though I think there should be ways to just observe the quota and not fire requests at all (+wait) once its exceeded. 

I also had this Idea to rewrite this whole project into a [GitHub integration](https://developer.github.com/apps/building-integrations/). Not sure about the quotas there, though :)

Right now, I don't have the time to start exploring either of these solutions, but I'd happy to see PR's and help with them.

**Comment by Xiphe** [comment]
released as `v0.3.1`
